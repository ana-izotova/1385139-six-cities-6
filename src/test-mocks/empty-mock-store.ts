import {AuthorizationStatus, Cities, FetchStatus, SortType, NameSpace} from "../const";
import {UserInitialStateTypes} from "../store/user/user-types";
import {AllOffersInitialStateTypes} from "../store/all-offers/all-offers-types";
import {SingleOfferInitialStateTypes} from "../store/single-offer/single-offer-types";
import {FavoritesInitialStateTypes} from "../store/favorites/favorites-types";

const userEmptyStateUnauthorized: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  login: ``,
  userAvatar: ``,
  fetchStatus: FetchStatus.INIT
};

const userStateAuthorized: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.AUTH,
  login: `jondoe@mail.com`,
  userAvatar: `avatar.jpg`,
  fetchStatus: FetchStatus.DONE
};

const allOffersEmptyStateLoaded: AllOffersInitialStateTypes = {
  currentCity: Cities[0],
  allOffers: [],
  currentSort: SortType.POPULAR,
  isDataLoaded: true,
  fetchStatus: FetchStatus.INIT,
  favoritesHaveBeenChanged: false,
  error: null
};

const allOffersEmptyState: AllOffersInitialStateTypes = {
  currentCity: Cities[0],
  allOffers: [],
  currentSort: SortType.POPULAR,
  isDataLoaded: false,
  fetchStatus: FetchStatus.INIT,
  favoritesHaveBeenChanged: false,
  error: null
};

const singleOfferEmptyState: SingleOfferInitialStateTypes = {
  offer: null,
  offersNearby: [],
  comments: [],
  isOfferLoaded: true,
  fetchStatus: FetchStatus.INIT,
  error: null
};

const favoritesEmptyState: FavoritesInitialStateTypes = {
  favoriteCards: [],
  areFavoriteCardsLoaded: false
};

const favoritesEmptyStateLoaded: FavoritesInitialStateTypes = {
  favoriteCards: [],
  areFavoriteCardsLoaded: true
};

export const emptyMockStoreUnauthorized = {
  [NameSpace.ALL_OFFERS]: allOffersEmptyState,
  [NameSpace.SINGLE_OFFER]: singleOfferEmptyState,
  [NameSpace.USER]: userEmptyStateUnauthorized,
  [NameSpace.FAVORITES]: favoritesEmptyState
};

export const emptyMockStoreAuthorized = {
  [NameSpace.ALL_OFFERS]: allOffersEmptyStateLoaded,
  [NameSpace.SINGLE_OFFER]: singleOfferEmptyState,
  [NameSpace.USER]: userStateAuthorized,
  [NameSpace.FAVORITES]: favoritesEmptyStateLoaded
};
