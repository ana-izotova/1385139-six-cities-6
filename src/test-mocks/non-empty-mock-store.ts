import {AuthorizationStatus, Cities, FetchStatus, NameSpace, SortType} from "../const";
import {UserInitialStateTypes} from "../store/user/user-types";
import {SingleOfferInitialStateTypes} from "../store/single-offer/single-offer-types";
import {currentOffer, offersNearby, comments, allOffers, favoriteOffers} from "./adapted-data-mock";
import {AllOffersInitialStateTypes} from "../store/all-offers/all-offers-types";
import {FavoritesInitialStateTypes} from "../store/favorites/favorites-types";

const userStoreAuthorized: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.AUTH,
  login: `jondoe@mail.com`,
  userAvatar: `avatar.jpg`,
  fetchStatus: FetchStatus.DONE
};

const userStoreUnauthorized: UserInitialStateTypes = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  login: ``,
  userAvatar: ``,
  fetchStatus: FetchStatus.DONE
};

const singleOfferStore: SingleOfferInitialStateTypes = {
  offer: currentOffer,
  offersNearby,
  comments,
  isOfferLoaded: true,
  fetchStatus: FetchStatus.DONE,
  error: null
};

const allOffersStore: AllOffersInitialStateTypes = {
  currentCity: Cities[0],
  allOffers,
  currentSort: SortType.POPULAR,
  isDataLoaded: true,
  fetchStatus: FetchStatus.DONE,
  favoritesHaveBeenChanged: true,
  error: null
};

const favoritesStore: FavoritesInitialStateTypes = {
  favoriteCards: favoriteOffers,
  areFavoriteCardsLoaded: true
};

export const nonEmptyMockStoreUnauthorized = {
  [NameSpace.ALL_OFFERS]: allOffersStore,
  [NameSpace.SINGLE_OFFER]: singleOfferStore,
  [NameSpace.USER]: userStoreUnauthorized,
  [NameSpace.FAVORITES]: favoritesStore
};

export const nonEmptyMockStoreAuthorized = {
  [NameSpace.ALL_OFFERS]: allOffersStore,
  [NameSpace.SINGLE_OFFER]: singleOfferStore,
  [NameSpace.USER]: userStoreAuthorized,
  [NameSpace.FAVORITES]: favoritesStore
};

