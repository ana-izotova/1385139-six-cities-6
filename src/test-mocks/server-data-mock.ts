export const testOffer = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 1,
};

export const testOffer2 = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": false,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 2,
};

export const testComment = {
  "id": 1,
  "user": {
    "id": 15,
    "is_pro": false,
    "name": `Kendall`,
    "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/6.jpg`,
  },
  "rating": 3,
  "comment": `We loved it so much, the house, the veiw, the location just great.. Highly recommend :)`,
  "date": `2021-02-12T08:04:28.647Z`,
};

export const testOfferFavorited = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": true,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 1,
};

export const testOffer2Favorited = {
  "city": {
    "name": `Paris`,
    "location": {
      "latitude": 48.85661,
      "longitude": 2.351499,
      "zoom": 13,
    },
  },
  "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`,
  "images": [
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/7.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/8.jpg`,
  ],
  "title": `The house among olive `,
  "is_favorite": true,
  "is_premium": true,
  "rating": 4,
  "type": `room`,
  "bedrooms": 2,
  "max_adults": 2,
  "price": 500,
  "goods": [`Breakfast`, `Air conditioning`],
  "host": {
    "id": 25,
    "name": `Madelina`,
    "is_pro": true,
    "avatar_url": `img/avatar-madelina.jpg`,
  },
  "description": `Relax, rejuvenate and unplug in this ultimate rustic getaway experience in the country.`,
  "location": {
    "latitude": 48.83861,
    "longitude": 2.350499,
    "zoom": 16,
  },
  "id": 2,
};


