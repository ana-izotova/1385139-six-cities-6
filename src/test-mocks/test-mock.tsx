import React, {Dispatch} from 'react';
import configureStore from 'redux-mock-store';
import * as redux from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import thunk, {ThunkDispatch} from 'redux-thunk';
import {RootStateType} from "../store/root-reducer";
import {ActionTypes} from "../store/action-types";
import {redirect} from "../store/middlewares/redirect";
import {emptyMockStoreAuthorized, emptyMockStoreUnauthorized} from "./empty-mock-store";
import {nonEmptyMockStoreUnauthorized, nonEmptyMockStoreAuthorized} from "./non-empty-mock-store";
import {AppThunk} from "../store/api-actions";

interface TestProps {
  children?: any,
  pushUrl?: string,
  emptyStore: boolean,
  authorized: boolean
}

export const TestMock: React.FC<TestProps> = ({children, pushUrl, emptyStore, authorized}) => {
  jest.spyOn(redux, `useDispatch`);
  jest.spyOn(redux, `useSelector`);

  const mockStore = configureStore<RootStateType, ThunkDispatch<RootStateType, AppThunk, ActionTypes>>([thunk, redirect]);

  let mock;
  if (emptyStore) {
    mock = authorized ? emptyMockStoreAuthorized : emptyMockStoreUnauthorized;
  } else {
    mock = authorized ? nonEmptyMockStoreAuthorized : nonEmptyMockStoreUnauthorized;
  }

  const store = mockStore(mock);
  const history = createMemoryHistory();

  if (pushUrl) {
    history.push(pushUrl);
  }

  return (
    <redux.Provider store={store}>
      <Router history={history}>
        {children}
      </Router>
    </redux.Provider>
  );
};


