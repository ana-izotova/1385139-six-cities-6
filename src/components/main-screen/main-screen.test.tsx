import React from 'react';
import {render} from '@testing-library/react';
import {TestMock} from "../../test-mocks/test-mock";
import MainScreen from "./main-screen";

describe(`Main screen should render correctly`, () => {
  it(`Main screen should show no offers correctly when empty store`, () => {
    const {container} = render(
        <TestMock authorized={true} emptyStore={true}>
          <MainScreen />
        </TestMock>
    );

    expect(container).toMatchSnapshot();
  });

  it(`Main screen should render correctly with offers`, () => {
    const {container} = render(
        <TestMock authorized={true} emptyStore={false}>
          <MainScreen />
        </TestMock>
    );

    expect(container).toMatchSnapshot();
  });
});


