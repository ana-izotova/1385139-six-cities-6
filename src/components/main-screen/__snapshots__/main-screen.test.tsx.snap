// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main screen should render correctly Main screen should render correctly with offers 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                src="img/logo.svg"
                style="width: 81px; height: 41px;"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                    style="background-image: url(avatar.jpg); border-radius: 50%;"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    jondoe@mail.com
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/"
                  style="margin-left: 10px;"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Exit
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index "
    >
      <h1
        class="visually-hidden"
      >
        Cities
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item tabs__item--active"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="cities"
      >
        <div
          class="cities__places-container container"
        >
          <section
            class="cities__places places"
          >
            <h2
              class="visually-hidden"
            >
              Places
            </h2>
            <b
              class="places__found"
            >
              1
               
              place
               to stay in 
              Paris
            </b>
            <form
              action="#"
              class="places__sorting"
              method="get"
            >
              <span
                class="places__sorting-caption"
              >
                Sort by
              </span>
              <span
                class="places__sorting-type"
                tabindex="0"
              >
                Popular
                <svg
                  class="places__sorting-arrow"
                  style="width: 7px; height: 4px;"
                >
                  <use
                    xlink:href="#icon-arrow-select"
                  />
                </svg>
              </span>
              <ul
                class="places__options places__options--custom"
              >
                <li
                  class="places__option places__option--active"
                  tabindex="0"
                >
                  Popular
                </li>
                <li
                  class="places__option "
                  tabindex="0"
                >
                  Price: low to high
                </li>
                <li
                  class="places__option "
                  tabindex="0"
                >
                  Price: high to low
                </li>
                <li
                  class="places__option "
                  tabindex="0"
                >
                  Top rated first
                </li>
              </ul>
            </form>
            <div
              class="cities__places-list places__list tabs__content"
            >
              <article
                class="cities__place-card place-card"
              >
                <div
                  class="place-card__mark"
                >
                  <span>
                    Premium
                  </span>
                </div>
                <div
                  class="cities__image-wrapper place-card__image-wrapper"
                >
                  <a
                    href="/offer/1"
                  >
                    <img
                      alt="Place image"
                      class="place-card__image"
                      height="200"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
                      width="260"
                    />
                  </a>
                </div>
                <div
                  class="place-card__info"
                >
                  <div
                    class="place-card__price-wrapper"
                  >
                    <div
                      class="place-card__price"
                    >
                      <b
                        class="place-card__price-value"
                      >
                        €
                        169
                      </b>
                      <span
                        class="place-card__price-text"
                      >
                        / night
                      </span>
                    </div>
                    <button
                      class="place-card__bookmark-button place-card__bookmark-button--active button"
                      type="button"
                    >
                      <svg
                        class="place-card__bookmark-icon"
                        height="19"
                        width="18"
                      >
                        <use
                          xlink:href="#icon-bookmark"
                        />
                      </svg>
                      <span
                        class="visually-hidden"
                      >
                        To bookmarks
                      </span>
                    </button>
                  </div>
                  <div
                    class="place-card__rating rating"
                  >
                    <div
                      class="place-card__stars rating__stars"
                    >
                      <span
                        style="width: 60%;"
                      />
                      <span
                        class="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <h2
                    class="place-card__name"
                  >
                    <a
                      href="/offer/1"
                    >
                      The house among olive 
                    </a>
                  </h2>
                  <p
                    class="place-card__type"
                  >
                    Room
                  </p>
                </div>
              </article>
            </div>
          </section>
          <div
            class="cities__right-section"
          >
            <section
              class="property__map map leaflet-container leaflet-grab leaflet-touch-drag"
              id="map"
              style="height: 100%; position: relative;"
              tabindex="0"
            >
              <div
                class="leaflet-pane leaflet-map-pane"
                style="left: 0px; top: 0px;"
              >
                <div
                  class="leaflet-pane leaflet-tile-pane"
                >
                  <div
                    class="leaflet-layer "
                    style="z-index: 1;"
                  >
                    <div
                      class="leaflet-tile-container leaflet-zoom-animated"
                      style="z-index: 18; left: 0px; top: 0px;"
                    >
                      <img
                        alt=""
                        class="leaflet-tile"
                        role="presentation"
                        src="https://b.basemaps.cartocdn.com/rastertiles/voyager/13/4149/2818.png"
                        style="width: 256px; height: 256px; left: -130px; top: -72px;"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="leaflet-pane leaflet-shadow-pane"
                />
                <div
                  class="leaflet-pane leaflet-overlay-pane"
                />
                <div
                  class="leaflet-pane leaflet-marker-pane"
                >
                  <img
                    alt=""
                    class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                    src="./img/pin.svg"
                    style="margin-left: -15px; margin-top: -15px; width: 30px; height: 30px; left: -5px; top: 159px; z-index: 159;"
                    tabindex="0"
                  />
                </div>
                <div
                  class="leaflet-pane leaflet-tooltip-pane"
                />
                <div
                  class="leaflet-pane leaflet-popup-pane"
                />
              </div>
              <div
                class="leaflet-control-container"
              >
                <div
                  class="leaflet-top leaflet-left"
                />
                <div
                  class="leaflet-top leaflet-right"
                />
                <div
                  class="leaflet-bottom leaflet-left"
                />
                <div
                  class="leaflet-bottom leaflet-right"
                >
                  <div
                    class="leaflet-control-attribution leaflet-control"
                  >
                    <a
                      href="https://leafletjs.com"
                      title="A JS library for interactive maps"
                    >
                      Leaflet
                    </a>
                     | © 
                    <a
                      href="https://www.openstreetmap.org/copyright"
                    >
                      OpenStreetMap
                    </a>
                     contributors © 
                    <a
                      href="https://carto.com/attributions"
                    >
                      CARTO
                    </a>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`Main screen should render correctly Main screen should show no offers correctly when empty store 1`] = `
<div>
  <div
    class="page page--gray page--main"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                src="img/logo.svg"
                style="width: 81px; height: 41px;"
              />
            </a>
          </div>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/favorites"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                    style="background-image: url(avatar.jpg); border-radius: 50%;"
                  />
                  <span
                    class="header__user-name user__name"
                  >
                    jondoe@mail.com
                  </span>
                </a>
              </li>
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/"
                  style="margin-left: 10px;"
                >
                  <span
                    class="header__user-name user__name"
                  >
                    Exit
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <main
      class="page__main page__main--index page__main--index-empty"
    >
      <h1
        class="visually-hidden"
      >
        Cities
      </h1>
      <div
        class="tabs"
      >
        <section
          class="locations container"
        >
          <ul
            class="locations__list tabs__list"
          >
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item tabs__item--active"
              >
                <span>
                  Paris
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Cologne
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Brussels
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Amsterdam
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Hamburg
                </span>
              </a>
            </li>
            <li
              class="locations__item"
            >
              <a
                class="locations__item-link tabs__item "
              >
                <span>
                  Dusseldorf
                </span>
              </a>
            </li>
          </ul>
        </section>
      </div>
      <div
        class="cities"
      >
        <div
          class="cities__places-container cities__places-container--empty container"
        >
          <section
            class="cities__no-places"
          >
            <div
              class="cities__status-wrapper tabs__content"
            >
              <b
                class="cities__status"
              >
                No places to stay available
              </b>
              <p
                class="cities__status-description"
              >
                We could not find any property available at the moment in 
                Paris
              </p>
            </div>
          </section>
          <div
            class="cities__right-section"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
